lib/acts_as_ar_query.rb

```diff
@@ -194,6 +194,35 @@ def append_hash_arg(symbol, *val)
     end
   end
 
+  def append_hash_array_arg(symbol, default, *val)
+    val = val.flatten.compact
+    val = val.first if val.size == 1 && val.first.kind_of?(Hash)
+    dup.tap do |r|
+      r.options[symbol] = merge_hash_or_array(r.options[symbol], val, default)
+    end
+  end
+
+  # @param a [Array, Hash]
+  # @param b [Array, Hash]
+  # @param default default value for conversion to a hash. e.g.: {} or "ASC"
+  def merge_hash_or_array(a, b, default = {})
+    if a.blank?
+      b
+    elsif b.empty?
```

Comment:
took me a while on this one.
`a` can be nil, but b can not.
But I like that change.